src/jit/codegeninterface.h

```diff
@@ -564,6 +565,199 @@ class CodeGenInterface
     //  Keeps track of how many bytes we've pushed on the processor's stack.
     unsigned genStackLevel;
 
+public:
+#ifdef USING_VARIABLE_LIVE_RANGE
+    //--------------------------------------------
+    //
+    // VariableLiveKeeper: Holds an array of "VariableLiveDescriptor", one for each variable
+    //  whose location we track. It provides start/end/update/count operations over the
+    //  "LiveRangeList" of any variable.
+    //
+    // Notes:
+    //  This method could be implemented on Compiler class too, but the intention is to move code
+    //  out of that class, which is huge. With this solution the only code needed in Compiler is
+    //  a getter and an initializer of this class.
+    //  The index of each variable in this array corresponds to the one in "compiler->lvaTable".
+    //  We care about tracking the variable locations of arguments, special arguments, and local IL
+    //  variables, and we ignore any other variable (like JIT temporary variables).
+    //
+    class VariableLiveKeeper
```

Comment:
Moving the pointer from compiler to codegen is good. Moving the class itself inside codegen seems excessive.

Reply:
@mikedn Where do you think it should be? I didn't want to make compiler.* files bigger.

Reply:
The file is fine as it is, I was just thinking that it shouldn't be a nested class because it is used in multiple places outside codegen.

But I looked again at the changes and I think it's fine currently. All usages outside codegen are like
```C++
compiler->codeGen->getVariableLiveKeeper()->si...
```
so at least for now it doesn't make any difference if it's nested or not. IMO it only matters if you have to write `CodeGenInterface::VariableLiveKeeper` multiple times to the point that it becomes a nuisance.